src/app/Recordings/Filters/DatetimeFilter.tsx

```diff
@@ -26,150 +27,239 @@ import {
   HelperTextItem,
   InputGroup,
   Popover,
-  Stack,
-  StackItem,
   TextInput,
   ValidatedOptions,
   InputGroupItem,
+  InputGroupText,
+  CalendarProps,
 } from '@patternfly/react-core';
 import { OutlinedCalendarAltIcon, SearchIcon } from '@patternfly/react-icons';
-import { t } from 'i18next';
 import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { DATETIME_INPUT_MAXLENGTH } from './const';
 
-export interface DateTimeFilterProps {
-  onSubmit: (dateISO: string) => void;
-}
-
-const _emptyDatetimeInput: {
+interface _DatetimeState {
   text: string;
   date: Date | undefined; // Ignore timezone
   validation: ValidatedOptions;
-} = {
+}
+
+const _emptyDatetimeInput: _DatetimeState = {
   text: '',
   date: undefined,
   validation: ValidatedOptions.default,
 };
 
+export interface DateTimeRange {
+  from?: string; // ISO format
+  to?: string; // ISO format
```

Comment:
This typecheck comment kind of scares me :-)

It looks like it would be worth defining a type like `DateTime`, which just looks roughly like:

```
class DateTime {
  private constructor(private readonly str: string) { }

  public toString(): string {
    return this.str;
  }

  public static of(str: string): DateTime {
    if (!validate(str)) {
      throw new Error();
    }
    return new DateTime(str);
  }
}
```

Basically, just a format-validated wrapper around a `string`, so that we can use the compiler for some build-time type safety to ensure runtime safety (assuming the `validate()` function is correct).

Reply:
Or, maybe there is an existing type like `Date` that could be suitable here instead, if there is some function that can take a `Date` and print it in ISO-8601 `string` format?

Reply:
> Basically, just a format-validated wrapper around a string, so that we can use the compiler for some build-time type safety to ensure runtime safety (assuming the validate() function is correct). This way, if we get some bad input, we have checked it early and failed at a well-known point where we have a nice validation function to tell us exactly what went wrong, rather than potentially failing somewhere deep in the call stack with strange error messages or just really unexpected behaviours.

Oh right, I think we could use `Date` type here. Though, within the Redux store, the filters will likely need to be stored as ISO strings as we persist the store data into Local Storage.

Reply:
That's fine, in that case the Redux<->LocalStorage interface is where the (de)serialization and ISO-8601 handling is performed, so it can be at that boundary where the Date<->string conversion is done.

Reply:
Sounds good! Fixed now. I added a `sanitize` func to convert ISO string to date when Redux store is loaded.